lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -99,7 +100,14 @@ def ssl_enabled(java_opts, credentials)
       def tier_name(java_opts, credentials)
         name = credentials['tier-name'] || @configuration['default_tier_name'] ||
           @application.details['application_name']
-        java_opts.add_system_property('appdynamics.agent.tierName', "#{name}")
+        @appdynamics_tier_name = ''
+        if @appdynamics_application_name == name
+          java_opts.add_system_property('appdynamics.agent.tierName', "#{name}-tier")
```

Comment:
For this change again why not just use an expression as the default that takes the app name that appends -tier?